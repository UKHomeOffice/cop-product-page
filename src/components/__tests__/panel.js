import React from "react";
import renderer from "react-test-renderer";
import Panel from "../panel";
import { render } from "@testing-library/react";

describe("Panel", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Panel />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with sign in button", () => {
    const { getByText } = render(<Panel />);
    const signInButton = getByText("Sign-in");
    const findOutMore = getByText("find out more about COP");
    expect(signInButton).toBeTruthy();
    expect(findOutMore).toBeTruthy();
  });
});
