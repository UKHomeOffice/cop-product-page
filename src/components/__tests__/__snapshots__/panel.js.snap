// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Array [
  <div
    className="govuk-panel"
  >
    <div
      className="govuk-width-container"
      style={
        Object {
          "marginTop": "30px",
        }
      }
    >
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <h1
            className="govuk-heading-xl panel-heading"
            style={
              Object {
                "color": "white",
              }
            }
          />
          <div
            className="banner-image-mobile"
          >
            <div
              style={
                Object {
                  "margin": "0 auto",
                  "maxWidth": "180px",
                }
              }
            >
              <img
                style={
                  Object {
                    "maxWidth": "100%",
                  }
                }
              />
            </div>
          </div>
          <p
            className="govuk-body govuk-!-font-size-24"
            style={
              Object {
                "color": "white",
              }
            }
          />
          <div
            className="button-container "
            style={
              Object {
                "marginBottom": "20px",
              }
            }
          >
            <a
              className="govuk-button product-page-button button-container__button govuk-!-margin-right-3"
              draggable="false"
              href="https://www.cop.homeoffice.gov.uk/"
              role="button"
            >
              <div
                style={
                  Object {
                    "paddingLeft": "20px",
                  }
                }
              >
                Sign-in 
                <i
                  className="sign-in-arrow"
                />
              </div>
            </a>
            or Â 
            <a
              className="govuk-link"
              href="/about/"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              find out more about COP
            </a>
          </div>
        </div>
        <div
          className="govuk-grid-column-one-third"
        >
          <div
            className="banner-image"
          >
            <img
              style={
                Object {
                  "maxWidth": "100%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="govuk-panel"
    id="sub-panel"
    style={
      Object {
        "background": "#bbd4fe",
        "margin": "0 auto",
        "paddingTop": "10px",
      }
    }
  >
    <div
      className="govuk-width-container"
    >
      <a
        className="boxclose govuk-link"
        id="boxclose"
        onClick={[Function]}
        onKeyPress={[Function]}
      />
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <p
            className="govuk-body-m"
            style={
              Object {
                "marginBottom": "0px",
              }
            }
          >
            <span
              className="govuk-!-font-weight-bold"
            >
              Mandatory Declarations (Man Dec)
            </span>
             submission deadline is 
            <span
              className="govuk-!-font-weight-bold"
            >
              31 April 2020
            </span>
            <br />
            <a
              className="govuk-link govuk-!-font-weight-bold"
              href="https://www.cop.homeoffice.gov.uk/"
            >
              Sign-in
            </a>
             to COP to fill the form, or 
            <a
              className="govuk-link govuk-!-font-weight-bold"
              href="https://training.cop.homeoffice.gov.uk/mandec.html#mandatory-declarations-man-dec"
            >
              find out more about the Man Dec.
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
